### PR TITLE
modify the judgment logic of httpMethodMatch

### PR DESCRIPTION
**Pull request by NingleXu**
## What is the purpose of the change
issues #12907 
### problem recurrence
image
When I use get requests, I can get the correct results
image
When I used the post request, I received an error message
image
### problem solving
At the beginning, I also thought it was because the httpMethod field did not add the hashCode method. But after thinking about it, I found that the httpMethod did not add the hashCode method, which only made the hashMap more prone to element collisions, not the reason for obtaining the wrong RestMethodMetadata.

It can be seen from the following figure that the wrong InvokerAndRestMethodMetadataPair is obtained from the pathToServiceMapNoPathVariable
image

Finally, I found that the httpMethodMatch method needs to determine whether the httpMethods are equal as long as one of the two PathMatchers needCompareHttpMethods is true. But its judgment is wrong.
image

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by Xzhhhhhh** [comment]
@AlbumenJ 能帮我CR一下吗

**Review by AlbumenJ — APPROVED** [review]
LGTM

**Comment by AlbumenJ** [comment]
@suncairong163 PTAL

**Comment by CrazyHZM** [comment]
@Xzhhhhhh Please fix unit test.

**Comment by Xzhhhhhh** [comment]
Fixing #12890  also fixes #12907 ， so I changed the comment on httpMethodMatch and added some unit tests to verify it in my pr. @suncairong163

**Comment by Xzhhhhhh** [comment]
> Would u pls add some uts or its to verify it?
ok, I added the unit tests and verified that the results were correct.

**Comment by AlbumenJ** [comment]
image

**Comment by Xzhhhhhh** [comment]
> image
ok, I forgot a detail, and it's been fixed

File: dubbo-metadata/dubbo-metadata-api/src/main/java/org/apache/dubbo/metadata/rest/PathMatcher.java [hunk 1]
```diff
@@ -174,7 +174,7 @@ && httpMethodMatch(that) // http method compare
     }
 
     /**
-     * it is needed to compare http method when one of needCompareHttpMethod is true,and don`t compare when both needCompareHttpMethod are false
+     * needCompareHttpMethod is not compared when one of them is false, and needCompareHttpMethod is compared when both are true
```
Comment (AlbumenJ) [inline dubbo-metadata/dubbo-metadata-api/src/main/java/org/apache/dubbo/metadata/rest/PathMatcher.java:177]: Is there any **real** changes?

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Pull request closed without merging**